homeassistant/components/template/fan.py

```diff
@@ -344,6 +345,9 @@ async def async_update(self):
         # Validate state
         if state in _VALID_STATES:
             self._state = state
+        elif state == STATE_UNAVAILABLE:
+            self._state = state
```

Comment:
We can remove this line since we set available below. It will take precedence over state.